docs/DeveloperGuide.md

```diff
@@ -2,37 +2,189 @@
+3. Add/Delete `Student` and `Component`.
+4. Return `CommandResult` that contains the result of the Add/Delete operation.
+
+The following sequence diagrams depict the exact steps involved in the `AddStudentCommand`:
+
+![AddStudentCommand.png](diagrams/AddStudentCommand.png)
+
+The logic for `AddComponentCommand` is very similar and thus is not depicted. 
+
+Likewise, the sequence diagram for `DeleteStudentCommand` is as follows: 
+
+![DeleteStudentCommand.png](diagrams%2FDeleteStudentCommand.png)
+
+Since a `Grade` object is only well-defined when there are both `Student` and `Component` objects to be refrenced by `Grade`,
+whenever a `Student` or `Component` object is deleted, the corresponding `Grade` object is queried and then deleted as well. 
+
+### Add/Delete Grade Feature
+
+#### Implementation Details
+
+The `AddGradeCommand` and `DeleteGradeCommand` classes handle the addition and deletion of grades for students within the TutorLink application. Each command validates user input to ensure accuracy and consistency before making changes, preserving data integrity. Grades are stored as `Grade` objects within a `GradeList`.
+
+#### Key Operations
+
+- **`AddGradeCommand.execute(AppState appState, HashMap<String, String> arguments)`**: Adds a grade to a student by performing the following steps:
+  1. Retrieves and validates the matriculation number, component description, and score from `arguments`.
+  2. Checks that the specified component and student exist.
+  3. Ensures the score is within the allowable range for the specified component.
+  4. Creates a new `Grade` object and adds it to the `GradeList` in `AppState`.
+
+- **`DeleteGradeCommand.execute(AppState appState, HashMap<String, String> arguments)`**: Removes a grade from a student by performing these steps:
+  1. Retrieves and validates the matriculation number and component description from `arguments`.
+  2. Confirms the existence of the specified component and student.
+  3. Locates and deletes the `Grade` object from the `GradeList` in `AppState`.
+
+#### Example Usage Scenario
+
+Given below is an example usage scenario and how the AddGradeCommand and DeleteGradeCommand behaves at each step.
+
+Consider a scenario where the Professor wants to add a grade for a student, Alice, in the midterm component of a module. The Professor would initiate the AddGradeCommand, inputting Alice's matriculation number, the component as "midterm," and a score of 85. The command will then:
+
+1. Validate Alice's matriculation number and ensure that the midterm component is present in the grade configuration for the module.
+2. Confirm that the score of 85 is within the allowable range for midterm.
+3. Create a new Grade object and add it to the GradeList in AppState, reflecting Alice's score for the midterm component.
+
+The sequence diagram of the DeleteGradeCommand is shown below.
+
+![AddGradeCommand.png](diagrams%2FAddGradeCommand.png)
```

Comment:
should AddStudentCommand have a X at the end of the lifeline of an object to show its deletion